tower-http/src/content_encoding.rs

```diff
@@ -201,32 +204,57 @@ pub(crate) fn encodings(
     headers: &http::HeaderMap,
     supported_encoding: impl SupportedEncodings,
 ) -> Vec<(Encoding, QValue)> {
-    headers
+    // Mapping of encodings to corresponding qvalues.
+    let mut encodings = std::collections::BTreeMap::default();
+
+    // Qvalue corresponding to the wildcard (`*`) encoding.
+    let mut wildcard_qval = None;
+
+    for v in headers
         .get_all(http::header::ACCEPT_ENCODING)
         .iter()
         .filter_map(|hval| hval.to_str().ok())
         .flat_map(|s| s.split(','))
-        .filter_map(|v| {
-            let mut v = v.splitn(2, ';');
+    {
+        let mut v = v.splitn(2, ';');
+        let encoding = v.next().unwrap().trim();
 
-            let encoding = match Encoding::parse(v.next().unwrap().trim(), supported_encoding) {
+        let qval = if let Some(qval) = v.next() {
+            if let Some(qval) = QValue::parse(qval.trim()) {
+                qval
+            } else {
+                continue; // ignore invalid qvalues
+            }
+        } else {
+            QValue::one()
+        };
+
+        if encoding == "*" {
+            wildcard_qval = Some(qval);
+        } else {
+            let encoding = match Encoding::parse(encoding, supported_encoding) {
                 Some(encoding) => encoding,
-                None => return None, // ignore unknown encodings
+                None => continue, // ignore unknown encodings
             };
 
-            let qval = if let Some(qval) = v.next() {
-                if let Some(qval) = QValue::parse(qval.trim()) {
-                    qval
-                } else {
-                    return None;
-                }
-            } else {
-                QValue::one()
-            };
+            encodings.insert(encoding, qval);
+        }
+    }
 
-            Some((encoding, qval))
-        })
-        .collect::<Vec<(Encoding, QValue)>>()
+    // The wildcard encoding (`*`) means all encodings not mentioned explicitly. If a wildcard
+    // encoding has been specified, set the qvalues of all unset encodings accordingly.
+    if wildcard_qval.is_some() {
+        let wildcard_qval = wildcard_qval.unwrap();
```

Comment:
Use `if let Some(wildcard_qval) = ...` instead to avoid unwrapping here`

Reply:
Done!